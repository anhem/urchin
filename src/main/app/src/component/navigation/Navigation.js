import React, {Component} from 'react'

class Navigation extends Component {

    toggleNavigation = () => {
        this.props.callbacks.toggleNavigation(this.props.navigationVisible)
    };

    render() {
        let toggleNav = this.props.navigationVisible ? 'navigation-menu--show' : 'navigation-menu--hide';

        return (
            <nav id="navigation">
                <a id="navigation__brand" href="/">Urchin</a>
                <button id="navigation--toggler" onClick={this.toggleNavigation}>
                    <span id="navigation-toggler__icon"/>
                </button>
                <div className={toggleNav} id="navigation-menu">
                    <ul id="navigation-menu__items">
                        <il className="navigation-item">
                            <a id="navigation-item__users" href="/users">Users</a>
                        </il>
                        <il className="navigation-item">
                            <a id="navigation-item__groups" href="/groups">Groups</a>
                        </il>
                        <il className="navigation-item">
                            <a id="navigation-item__folders" href="/folders">Folders</a>
                        </il>
                    </ul>
                </div>

            </nav>
        )
    }
}

export default Navigation