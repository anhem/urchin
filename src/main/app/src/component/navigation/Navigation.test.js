import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import Navigation from "./Navigation";

describe('Navigation', () => {

    const callbacks = {
        toggleNavigation: jest.fn(),
    };

    const props = {
        navigationVisible: false,
        callbacks: callbacks
    };

    let component;

    beforeEach(() => {
        Object.entries(callbacks).forEach(([name, fn]) => {
            fn.mockClear();
        });
        component = shallow(<Navigation {...props}/>);
    });

    it('match snapshot', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('toggleNavigation is called when clicking toggleNavigation', () => {
        component.find('[data-view="toggleNavigation"]').simulate('click');

        expect(callbacks.toggleNavigation).toHaveBeenCalledTimes(1)
    });

    it('toggleNavigation is called when visible while clicking navigation item', () => {
        let props = {
            navigationVisible: true,
            callbacks: callbacks
        };
        component = shallow(<Navigation {...props}/>);

        component.find('[data-view="navigationBrand"]').simulate('click');

        expect(callbacks.toggleNavigation).toHaveBeenCalledTimes(1)
    });

    it('toggleNavigation is not called when not visible while clicking navigation item', () => {
        component.find('[data-view="navigationBrand"]').simulate('click');

        expect(callbacks.toggleNavigation).toHaveBeenCalledTimes(0)
    });
});
