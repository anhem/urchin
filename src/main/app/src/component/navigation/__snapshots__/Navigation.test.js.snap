// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation match snapshot 1`] = `
<nav
  id="navigation"
>
  <a
    href="/"
    id="navigation__brand"
  >
    Urchin
  </a>
  <button
    id="navigation--toggler"
    onClick={[Function]}
  >
    <span
      id="navigation-toggler__icon"
    />
  </button>
  <div
    className="navigation-menu--hide"
    id="navigation-menu"
  >
    <ul
      id="navigation-menu__items"
    >
      <il
        className="navigation-item"
      >
        <a
          href="/users"
          id="navigation-item__users"
        >
          Users
        </a>
      </il>
      <il
        className="navigation-item"
      >
        <a
          href="/groups"
          id="navigation-item__groups"
        >
          Groups
        </a>
      </il>
      <il
        className="navigation-item"
      >
        <a
          href="/folders"
          id="navigation-item__folders"
        >
          Folders
        </a>
      </il>
    </ul>
  </div>
</nav>
`;
